sentry-ruby/lib/sentry/transport.rb

```diff
@@ -45,29 +45,22 @@ def generate_auth_header
       'Sentry ' + fields.map { |key, value| "#{key}=#{value}" }.join(', ')
     end
 
-    def encode(event_hash)
-      event_id = event_hash[:event_id] || event_hash['event_id']
-      event_type = event_hash[:type] || event_hash['type']
+    def encode(event)
+      # Convert to hash
+      event_hash = event.to_hash
+
+      event_id = event_hash[:event_id] || event_hash["event_id"]
+      envelope_type = event_hash[:type] || event_hash["type"] || "event"
```

Comment:
This is the type of the item in the envelope, not the type of the envelope itself. There could, for instance, be another item of type `"attachment"`. See [here](https://develop.sentry.dev/sdk/envelopes/#data-model).

```suggestion
      item_type = event_hash[:type] || event_hash["type"] || "event"
```

Reply:
👍 updated